ccl/multiregionccl: TestMultiRegionDataDriven_global_tables failed

ccl/multiregionccl.TestMultiRegionDataDriven_global_tables [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20360623?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20360623?buildTab=artifacts#/) on master @ [7e7425f698b4f612a2e72cc8e813b44c52af89cb](https://github.com/cockroachdb/cockroach/commits/7e7425f698b4f612a2e72cc8e813b44c52af89cb):


```
               4.251ms      0.009ms                            === operation:/cockroach.roachpb.Internal/Batch gid:73302 _verbose:1 node:5 span.kind:server request:Get [/Table/108/1/1/0], [txn: a8068938], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 46429]
               4.261ms      0.010ms                            event:server/node.go:1620 [n5] node received request: 1 Get
               4.289ms      0.029ms                            event:kv/kvserver/store_send.go:149 [n5,s5] executing Get [/Table/108/1/1/0], [txn: a8068938], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 46429]
               4.303ms      0.014ms                            event:kv/kvserver/replica_send.go:183 [n5,s5,r85/6:/{Table/108-Max}] read-only path
               4.317ms      0.013ms                            event:kv/kvserver/concurrency/concurrency_manager.go:259 [n5,s5,r85/6:/{Table/108-Max}] sequencing request
               4.327ms      0.010ms                            event:kv/kvserver/concurrency/concurrency_manager.go:340 [n5,s5,r85/6:/{Table/108-Max}] acquiring latches
               4.338ms      0.011ms                            event:kv/kvserver/concurrency/concurrency_manager.go:384 [n5,s5,r85/6:/{Table/108-Max}] scanning lock table for conflicting locks
               4.379ms      0.040ms                            event:kv/kvserver/replica_follower_read.go:133 [n5,s5,r85/6:/{Table/108-Max}] can't serve follower read; closed timestamp too low by: 198.852459ms; maxClosed: 1755614429.172327314,0 ts: 1755614428.871179773,0 uncertaintyLimit: 1755614429.371179773,0
               4.470ms      0.091ms                            event:kv/kvserver/replica_send.go:202 [n5,s5,r85/6:/{Table/108-Max}] replica.Send got error: [NotLeaseHolderError] lease held by different store; r85: replica (n5,s5):6NON_VOTER not lease holder; current lease is repl=(n1,s1):8 seq=3 start=1755614423.388749526,0 exp=1755614432.740241373,0 pro=1755614426.740241373,0 acq=Request
               4.536ms      0.067ms                            event:server/node.go:1677 [n5] error from stores.Send: [NotLeaseHolderError] lease held by different store; r85: replica (n5,s5):6NON_VOTER not lease holder; current lease is repl=(n1,s1):8 seq=3 start=1755614423.388749526,0 exp=1755614432.740241373,0 pro=1755614426.740241373,0 acq=Request: "sql txn" meta={id=a8068938 key=/Min iso=Serializable pri=0.00054007 epo=0 ts=1755614428.871179773,0 min=1755614428.871179773,0 seq=0} lock=false stat=PENDING rts=1755614428.871179773,0 gul=1755614429.371179773,0 obs={n5@1755614428.871179773,0}
               4.549ms      0.013ms                            event:server/node.go:1714 [n5] node sending response
               4.586ms      0.344ms                        event:kv/kvclient/kvcoord/transport.go:212 [n5,client=127.0.0.1:48398,hostssl,user=root,txn=a8068938] received batch response
               4.650ms      0.064ms                        event:kv/kvclient/kvcoord/dist_sender.go:2909 [n5,client=127.0.0.1:48398,hostssl,user=root,txn=a8068938] [NotLeaseHolderError] lease held by different store; r85: replica (n5,s5):6NON_VOTER not lease holder; current lease is repl=(n1,s1):8 seq=3 start=1755614423.388749526,0 exp=1755614432.740241373,0 pro=1755614426.740241373,0 acq=Request: "sql txn" meta={id=a8068938 key=/Min iso=Serializable pri=0.00054007 epo=0 ts=1755614428.871179773,0 min=1755614428.871179773,0 seq=0} lock=false stat=PENDING rts=1755614428.871179773,0 gul=1755614429.371179773,0 obs={n5@1755614428.871179773,0}
               4.736ms      0.085ms                        event:kv/kvclient/kvcoord/dist_sender.go:3153 [n5,client=127.0.0.1:48398,hostssl,user=root,txn=a8068938] application error: [NotLeaseHolderError] lease held by different store; r85: replica (n5,s5):6NON_VOTER not lease holder; current lease is repl=(n1,s1):8 seq=3 start=1755614423.388749526,0 exp=1755614432.740241373,0 pro=1755614426.740241373,0 acq=Request: "sql txn" meta={id=a8068938 key=/Min iso=Serializable pri=0.00054007 epo=0 ts=1755614428.871179773,0 min=1755614428.871179773,0 seq=0} lock=false stat=PENDING rts=1755614428.871179773,0 gul=1755614429.371179773,0 obs={n5@1755614428.871179773,0}
               4.764ms      0.028ms                        event:kv/kvclient/kvcoord/dist_sender.go:2696 [n5,client=127.0.0.1:48398,hostssl,user=root,txn=a8068938] trying next peer (n1,s1):8
               4.778ms      0.014ms                        event:rpc/nodedialer/nodedialer.go:230 [n5,client=127.0.0.1:48398,hostssl,user=root,txn=a8068938] sending request to 127.0.0.1:41135
               4.793ms      0.015ms                        event:kv/kvclient/kvcoord/transport.go:210 [n5,client=127.0.0.1:48398,hostssl,user=root,txn=a8068938] sending batch request
               4.795ms      0.002ms                            === operation:/cockroach.roachpb.Internal/Batch gid:73302 _verbose:1 node:5 client:127.0.0.1:48398 hostssl: user:root txn:a8068938 span.kind:client
              11.644ms      6.849ms                            === operation:/cockroach.roachpb.Internal/Batch gid:74433 _verbose:1 node:1 span.kind:server request:Get [/Table/108/1/1/0], [txn: a8068938], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 46429]
              11.667ms      0.023ms                            event:server/node.go:1620 [n1] node received request: 1 Get
              11.701ms      0.034ms                            event:kv/kvserver/store_send.go:149 [n1,s1] executing Get [/Table/108/1/1/0], [txn: a8068938], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 46429]
              11.718ms      0.017ms                            event:kv/kvserver/replica_send.go:183 [n1,s1,r85/8:/{Table/108-Max}] read-only path
              11.731ms      0.013ms                            event:kv/kvserver/concurrency/concurrency_manager.go:259 [n1,s1,r85/8:/{Table/108-Max}] sequencing request
              11.740ms      0.009ms                            event:kv/kvserver/concurrency/concurrency_manager.go:340 [n1,s1,r85/8:/{Table/108-Max}] acquiring latches
              11.751ms      0.011ms                            event:kv/kvserver/concurrency/concurrency_manager.go:384 [n1,s1,r85/8:/{Table/108-Max}] scanning lock table for conflicting locks
              11.789ms      0.038ms                            event:kv/kvserver/replica_read.go:316 [n1,s1,r85/8:/{Table/108-Max}] can drop latches early for batch (Get [/Table/108/1/1/0], [txn: a8068938], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 46429]); scanning lock table first to detect conflicts
              11.822ms      0.033ms                            event:storage/engine.go:2073 [n1,s1,r85/8:/{Table/108-Max}] lock table scan stats: seeked 1 times (1 internal); stepped 0 times (0 internal)
              11.833ms      0.011ms                            event:kv/kvserver/replica_read.go:120 [n1,s1,r85/8:/{Table/108-Max}] lock table scan complete without conflicts; dropping latches early
              11.849ms      0.016ms                            event:kv/kvserver/replica_read.go:429 [n1,s1,r85/8:/{Table/108-Max}] executing read-only batch
              11.941ms      0.092ms                            event:kv/kvserver/replica_evaluate.go:567 [n1,s1,r85/8:/{Table/108-Max}] evaluated Get command header:<key:"\364\211\211\210" > expect_exclusion_since:<> , txn="sql txn" meta={id=a8068938 key=/Min iso=Serializable pri=0.00054007 epo=0 ts=1755614428.871179773,0 min=1755614428.871179773,0 seq=0} lock=false stat=PENDING rts=1755614428.871179773,0 gul=1755614429.371179773,0 obs={n1@1755614428.883212219,0 n5@1755614428.871179773,0} : resp=header:<num_keys:1 num_bytes:7 > value:<raw_bytes:"\375\311[\010\n#\002" timestamp:<wall_time:1755614409847570594 logical:2 > > , err=<nil>
              11.966ms      0.025ms                            event:n1 scan stats: stepped 0 times (0 internal); seeked 1 times (1 internal); block-bytes: (total 0 B, cached 0 B, duration 0s); points: (count 1, key-bytes 18 B, value-bytes 7 B, tombstoned: 0) ranges: (count 0), (contained-points 1, skipped-points 0) evaluated requests: 1 gets, 0 scans, 0 reverse scans
              11.978ms      0.012ms                            event:kv/kvserver/replica_read.go:241 [n1,s1,r85/8:/{Table/108-Max}] read completed
              11.997ms      0.019ms                            event:server/node.go:1714 [n1] node sending response
              11.944ms     -0.053ms                            structured:{"@type":"type.googleapis.com/cockroach.roachpb.ScanStats","numInterfaceSeeks":"1","numInternalSeeks":"1","keyBytes":"18","valueBytes":"7","pointCount":"1","rangeKeyContainedPoints":"1","numGets":"1","nodeId":1,"region":"us-east-1"}
             116.265ms    111.472ms                        event:kv/kvclient/kvcoord/transport.go:212 [n5,client=127.0.0.1:48398,hostssl,user=root,txn=a8068938] received batch response
             116.406ms    112.358ms                    event:kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:604 [n5,client=127.0.0.1:48398,hostssl,user=root,txn=a8068938] recording span to refresh: /Table/108/1/1/0
             116.529ms    112.531ms            event:ComponentStats{ID: {d2c932d1-d449-4fe9-9db8-899eed7f27f0 PROCESSOR 0 5 }, KV nodes: n1, KV regions: us-east-1, KV time: 112ms, KV rows decoded: 1, KV bytes read: 11 B, KV gRPC calls: 1, KV pairs read: 1, MVCC step count (ext/int): 0/0, MVCC seek count (ext/int): 1/1, max memory allocated: 20 KiB, sql cpu time: 22µs, batches output: 1, rows output: 1}
             116.538ms      0.009ms            event:ComponentStats{ID: {00000000-0000-0000-0000-000000000000 UNSET 0 ? }}
             116.493ms     -0.045ms            structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"d2c932d1-d449-4fe9-9db8-899eed7f27f0","type":"PROCESSOR","sqlInstanceId":5},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{"valuePlusOne":"12"},"kvPairsRead":{"valuePlusOne":"2"},"tuplesRead":{"valuePlusOne":"2"},"batchRequestsIssued":{"valuePlusOne":"2"},"kvTime":{"valuePlusOne":"0.112465570s"},"contentionTime":{"valuePlusOne":"0.000000001s"},"lockWaitTime":{"valuePlusOne":"0.000000001s"},"latchWaitTime":{"valuePlusOne":"0.000000001s"},"numInterfaceSteps":{"valuePlusOne":"1"},"numInternalSteps":{"valuePlusOne":"1"},"numInterfaceSeeks":{"valuePlusOne":"2"},"numInternalSeeks":{"valuePlusOne":"2"},"blockBytes":{"valuePlusOne":"1"},"blockBytesInCache":{"valuePlusOne":"1"},"keyBytes":{"valuePlusOne":"19"},"valueBytes":{"valuePlusOne":"8"},"pointCount":{"valuePlusOne":"2"},"pointsCoveredByRangeTombstones":{"valuePlusOne":"1"},"rangeKeyCount":{"valuePlusOne":"1"},"rangeKeyContainedPoints":{"valuePlusOne":"2"},"rangeKeySkippedPoints":{"valuePlusOne":"1"},"kvCpuTime":{},"numGets":{"valuePlusOne":"2"},"numScans":{"valuePlusOne":"1"},"numReverseScans":{"valuePlusOne":"1"},"nodeIds":[1],"regions":["us-east-1"]},"exec":{"execTime":{},"maxAllocatedMem":{"valuePlusOne":"20481"},"maxAllocatedDisk":{},"consumedRu":{"valuePlusOne":"1"},"cpuTime":{"valuePlusOne":"0.000021967s"}},"output":{"numBatches":{"valuePlusOne":"2"},"numTuples":{"valuePlusOne":"2"}},"flowStats":{"maxMemUsage":{},"maxDiskUsage":{},"consumedRu":{}}}
             116.533ms      0.040ms            structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"00000000-0000-0000-0000-000000000000"},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{},"kvPairsRead":{},"tuplesRead":{},"batchRequestsIssued":{},"kvTime":{},"contentionTime":{},"lockWaitTime":{},"latchWaitTime":{},"numInterfaceSteps":{},"numInternalSteps":{},"numInterfaceSeeks":{},"numInternalSeeks":{},"blockBytes":{},"blockBytesInCache":{},"keyBytes":{},"valueBytes":{},"pointCount":{},"pointsCoveredByRangeTombstones":{},"rangeKeyCount":{},"rangeKeyContainedPoints":{},"rangeKeySkippedPoints":{},"kvCpuTime":{},"numGets":{},"numScans":{},"numReverseScans":{}},"exec":{"execTime":{},"maxAllocatedMem":{},"maxAllocatedDisk":{},"consumedRu":{},"cpuTime":{}},"output":{"numBatches":{},"numTuples":{}},"flowStats":{"maxMemUsage":{},"maxDiskUsage":{},"consumedRu":{}}}
             116.636ms    112.638ms        event:ComponentStats{ID: {d2c932d1-d449-4fe9-9db8-899eed7f27f0 FLOW 0 5 us-central-1}}
             116.621ms     -0.015ms        structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"d2c932d1-d449-4fe9-9db8-899eed7f27f0","type":"FLOW","sqlInstanceId":5,"region":"us-central-1"},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{},"kvPairsRead":{},"tuplesRead":{},"batchRequestsIssued":{},"kvTime":{},"contentionTime":{},"lockWaitTime":{},"latchWaitTime":{},"numInterfaceSteps":{},"numInternalSteps":{},"numInterfaceSeeks":{},"numInternalSeeks":{},"blockBytes":{},"blockBytesInCache":{},"keyBytes":{},"valueBytes":{},"pointCount":{},"pointsCoveredByRangeTombstones":{},"rangeKeyCount":{},"rangeKeyContainedPoints":{},"rangeKeySkippedPoints":{},"kvCpuTime":{},"numGets":{},"numScans":{},"numReverseScans":{}},"exec":{"execTime":{},"maxAllocatedMem":{},"maxAllocatedDisk":{},"consumedRu":{},"cpuTime":{}},"output":{"numBatches":{},"numTuples":{}},"flowStats":{"maxMemUsage":{"valuePlusOne":"20481"},"maxDiskUsage":{"valuePlusOne":"1"},"consumedRu":{}}}
             116.704ms    112.789ms    event:sql/conn_executor_exec.go:1102 [n5,client=127.0.0.1:48398,hostssl,user=root] execution ends
             116.724ms      0.021ms    event:sql/conn_executor_exec.go:1102 [n5,client=127.0.0.1:48398,hostssl,user=root] rows affected: 1
             116.807ms      0.083ms        === operation:commit sql txn gid:73302 _verbose:1 node:5 client:127.0.0.1:48398 hostssl: user:root
             116.893ms      0.168ms    event:sql/conn_executor_exec.go:4274 [n5,client=127.0.0.1:48398,hostssl,user=root] AutoCommit. err: <nil>
        
    panic.go:636: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1a0d7b7d86f8cdff97782c5fadc5ffe4/logTestMultiRegionDataDriven_global_tables3208943664
--- FAIL: TestMultiRegionDataDriven_global_tables (64.99s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #150718 ccl/multiregionccl: TestMultiRegionDataDriven_global_tables failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.1]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven_global_tables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-53645

ccl/multiregionccl.TestMultiRegionDataDriven_global_tables [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20472921?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20472921?buildTab=artifacts#/) on master @ [1572ff3199f6c9d1971034a5ee8de6258b741864](https://github.com/cockroachdb/cockroach/commits/1572ff3199f6c9d1971034a5ee8de6258b741864):


```
               3.525ms      0.034ms                        event:kv/kvclient/kvcoord/transport.go:210 [n3,client=127.0.0.1:41026,hostssl,user=root,txn=8ac18761] sending batch request
               3.534ms      0.008ms                            === operation:/cockroach.roachpb.Internal/Batch gid:83002 _verbose:1 node:3 span.kind:server request:Get [/Table/108/1/1/0], [txn: 8ac18761], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]
               3.561ms      0.027ms                            event:server/node.go:1620 [n3] node received request: 1 Get
               3.619ms      0.058ms                            event:kv/kvserver/store_send.go:149 [n3,s3] executing Get [/Table/108/1/1/0], [txn: 8ac18761], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]
               3.648ms      0.029ms                            event:kv/kvserver/replica_send.go:183 [n3,s3,r88/2:/{Table/108-Max}] read-only path
               5.036ms      1.388ms                            event:kv/kvserver/concurrency/concurrency_manager.go:259 [n3,s3,r88/2:/{Table/108-Max}] sequencing request
               5.063ms      0.028ms                            event:kv/kvserver/concurrency/concurrency_manager.go:340 [n3,s3,r88/2:/{Table/108-Max}] acquiring latches
               5.094ms      0.031ms                            event:kv/kvserver/concurrency/concurrency_manager.go:384 [n3,s3,r88/2:/{Table/108-Max}] scanning lock table for conflicting locks
               5.161ms      0.066ms                            event:kv/kvserver/replica_follower_read.go:133 [n3,s3,r88/2:/{Table/108-Max}] can't serve follower read; closed timestamp too low by: 7.973242ms; maxClosed: 1758044106.557601910,0 ts: 1758044106.065575152,0 uncertaintyLimit: 1758044106.565575152,0
               6.587ms      1.426ms                            event:kv/kvserver/replica_send.go:202 [n3,s3,r88/2:/{Table/108-Max}] replica.Send got error: [NotLeaseHolderError] lease held by different store; r88: replica (n3,s3):2 not lease holder; current lease is repl=(n1,s1):8 seq=3 start=1758044103.144079438,0 exp=1758044109.144029111,0 pro=1758044103.144029111,0 acq=Transfer
               6.862ms      0.275ms                            event:server/node.go:1688 [n3] error from stores.Send: [NotLeaseHolderError] lease held by different store; r88: replica (n3,s3):2 not lease holder; current lease is repl=(n1,s1):8 seq=3 start=1758044103.144079438,0 exp=1758044109.144029111,0 pro=1758044103.144029111,0 acq=Transfer: "sql txn" meta={id=8ac18761 key=/Min iso=Serializable pri=0.01632539 epo=0 ts=1758044106.065575152,0 min=1758044106.065575152,0 seq=0} lock=false stat=PENDING rts=1758044106.065575152,0 gul=1758044106.565575152,0 obs={n3@1758044106.065575152,0}
               6.891ms      0.028ms                            event:server/node.go:1725 [n3] node sending response
               6.971ms      3.446ms                        event:kv/kvclient/kvcoord/transport.go:212 [n3,client=127.0.0.1:41026,hostssl,user=root,txn=8ac18761] received batch response
               7.107ms      0.136ms                        event:kv/kvclient/kvcoord/dist_sender.go:2878 [n3,client=127.0.0.1:41026,hostssl,user=root,txn=8ac18761] [NotLeaseHolderError] lease held by different store; r88: replica (n3,s3):2 not lease holder; current lease is repl=(n1,s1):8 seq=3 start=1758044103.144079438,0 exp=1758044109.144029111,0 pro=1758044103.144029111,0 acq=Transfer: "sql txn" meta={id=8ac18761 key=/Min iso=Serializable pri=0.01632539 epo=0 ts=1758044106.065575152,0 min=1758044106.065575152,0 seq=0} lock=false stat=PENDING rts=1758044106.065575152,0 gul=1758044106.565575152,0 obs={n3@1758044106.065575152,0}
               7.244ms      0.137ms                        event:kv/kvclient/kvcoord/dist_sender.go:3122 [n3,client=127.0.0.1:41026,hostssl,user=root,txn=8ac18761] application error: [NotLeaseHolderError] lease held by different store; r88: replica (n3,s3):2 not lease holder; current lease is repl=(n1,s1):8 seq=3 start=1758044103.144079438,0 exp=1758044109.144029111,0 pro=1758044103.144029111,0 acq=Transfer: "sql txn" meta={id=8ac18761 key=/Min iso=Serializable pri=0.01632539 epo=0 ts=1758044106.065575152,0 min=1758044106.065575152,0 seq=0} lock=false stat=PENDING rts=1758044106.065575152,0 gul=1758044106.565575152,0 obs={n3@1758044106.065575152,0}
               7.286ms      0.042ms                        event:kv/kvclient/kvcoord/dist_sender.go:2665 [n3,client=127.0.0.1:41026,hostssl,user=root,txn=8ac18761] trying next peer (n1,s1):8
               7.324ms      0.039ms                        event:rpc/nodedialer/nodedialer.go:230 [n3,client=127.0.0.1:41026,hostssl,user=root,txn=8ac18761] sending request to 127.0.0.1:42311
               7.354ms      0.030ms                        event:kv/kvclient/kvcoord/transport.go:210 [n3,client=127.0.0.1:41026,hostssl,user=root,txn=8ac18761] sending batch request
               7.448ms      0.094ms                            === operation:/cockroach.roachpb.Internal/Batch gid:82086 _verbose:1 node:1 span.kind:server request:Get [/Table/108/1/1/0], [txn: 8ac18761], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]
               7.498ms      0.050ms                            event:server/node.go:1620 [n1] node received request: 1 Get
               7.560ms      0.063ms                            event:kv/kvserver/store_send.go:149 [n1,s1] executing Get [/Table/108/1/1/0], [txn: 8ac18761], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]
               7.602ms      0.042ms                            event:kv/kvserver/replica_send.go:183 [n1,s1,r88/8:/{Table/108-Max}] read-only path
               8.088ms      0.486ms                            event:kv/kvserver/concurrency/concurrency_manager.go:259 [n1,s1,r88/8:/{Table/108-Max}] sequencing request
               8.405ms      0.317ms                            event:kv/kvserver/concurrency/concurrency_manager.go:340 [n1,s1,r88/8:/{Table/108-Max}] acquiring latches
               8.440ms      0.034ms                            event:kv/kvserver/concurrency/concurrency_manager.go:384 [n1,s1,r88/8:/{Table/108-Max}] scanning lock table for conflicting locks
               8.591ms      0.151ms                            event:kv/kvserver/replica_read.go:317 [n1,s1,r88/8:/{Table/108-Max}] can drop latches early for batch (Get [/Table/108/1/1/0], [txn: 8ac18761], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]); scanning lock table first to detect conflicts
               8.658ms      0.067ms                            event:storage/engine.go:2073 [n1,s1,r88/8:/{Table/108-Max}] lock table scan stats: seeked 1 times (1 internal); stepped 0 times (0 internal)
               8.687ms      0.029ms                            event:kv/kvserver/replica_read.go:120 [n1,s1,r88/8:/{Table/108-Max}] lock table scan complete without conflicts; dropping latches early
               8.723ms      0.036ms                            event:kv/kvserver/replica_read.go:430 [n1,s1,r88/8:/{Table/108-Max}] executing read-only batch
               8.962ms      0.239ms                            event:kv/kvserver/replica_evaluate.go:567 [n1,s1,r88/8:/{Table/108-Max}] evaluated Get command header:<key:"\364\211\211\210" > expect_exclusion_since:<> , txn="sql txn" meta={id=8ac18761 key=/Min iso=Serializable pri=0.01632539 epo=0 ts=1758044106.065575152,0 min=1758044106.065575152,0 seq=0} lock=false stat=PENDING rts=1758044106.065575152,0 gul=1758044106.565575152,0 obs={n1@1758044106.073155322,0 n3@1758044106.065575152,0} : resp=header:<num_keys:1 num_bytes:7 > value:<raw_bytes:"\375\311[\010\n#\002" timestamp:<wall_time:1758044083750115349 > > , err=<nil>
               9.003ms      0.041ms                            event:n1 scan stats: stepped 0 times (0 internal); seeked 1 times (1 internal); block-bytes: (total 176 B, cached 176 B, duration 0s); points: (count 1, key-bytes 14 B, value-bytes 7 B, tombstoned: 0) ranges: (count 0), (contained-points 0, skipped-points 0) evaluated requests: 1 gets, 0 scans, 0 reverse scans
               9.030ms      0.028ms                            event:kv/kvserver/replica_read.go:242 [n1,s1,r88/8:/{Table/108-Max}] read completed
               9.093ms      0.063ms                            event:server/node.go:1725 [n1] node sending response
               8.966ms     -0.127ms                            structured:{"@type":"type.googleapis.com/cockroach.roachpb.ScanStats","numInterfaceSeeks":"1","numInternalSeeks":"1","blockBytes":"176","blockBytesInCache":"176","keyBytes":"14","valueBytes":"7","pointCount":"1","numGets":"1","nodeId":1,"region":"us-east-1"}
               9.342ms      1.988ms                        event:kv/kvclient/kvcoord/transport.go:212 [n3,client=127.0.0.1:41026,hostssl,user=root,txn=8ac18761] received batch response
              11.670ms      8.560ms                    event:kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:604 [n3,client=127.0.0.1:41026,hostssl,user=root,txn=8ac18761] recording span to refresh: /Table/108/1/1/0
              11.791ms      8.790ms            event:ComponentStats{ID: {3449d711-580b-425b-9508-e7d4251346ff PROCESSOR 0 3 }, KV nodes: n1, KV regions: us-east-1, KV time: 9ms, KV rows decoded: 1, KV bytes read: 11 B, KV gRPC calls: 1, KV pairs read: 1, MVCC step count (ext/int): 0/0, MVCC seek count (ext/int): 1/1, max memory allocated: 20 KiB, sql cpu time: 22µs, batches output: 1, rows output: 1}
              11.812ms      0.021ms            event:ComponentStats{ID: {00000000-0000-0000-0000-000000000000 UNSET 0 ? }}
              11.729ms     -0.083ms            structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"3449d711-580b-425b-9508-e7d4251346ff","type":"PROCESSOR","sqlInstanceId":3},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{"valuePlusOne":"12"},"kvPairsRead":{"valuePlusOne":"2"},"tuplesRead":{"valuePlusOne":"2"},"batchRequestsIssued":{"valuePlusOne":"2"},"kvTime":{"valuePlusOne":"0.008699096s"},"contentionTime":{"valuePlusOne":"0.000000001s"},"lockWaitTime":{"valuePlusOne":"0.000000001s"},"latchWaitTime":{"valuePlusOne":"0.000000001s"},"numInterfaceSteps":{"valuePlusOne":"1"},"numInternalSteps":{"valuePlusOne":"1"},"numInterfaceSeeks":{"valuePlusOne":"2"},"numInternalSeeks":{"valuePlusOne":"2"},"blockBytes":{"valuePlusOne":"177"},"blockBytesInCache":{"valuePlusOne":"177"},"keyBytes":{"valuePlusOne":"15"},"valueBytes":{"valuePlusOne":"8"},"pointCount":{"valuePlusOne":"2"},"pointsCoveredByRangeTombstones":{"valuePlusOne":"1"},"rangeKeyCount":{"valuePlusOne":"1"},"rangeKeyContainedPoints":{"valuePlusOne":"1"},"rangeKeySkippedPoints":{"valuePlusOne":"1"},"kvCpuTime":{},"numGets":{"valuePlusOne":"2"},"numScans":{"valuePlusOne":"1"},"numReverseScans":{"valuePlusOne":"1"},"nodeIds":[1],"regions":["us-east-1"]},"exec":{"execTime":{},"maxAllocatedMem":{"valuePlusOne":"20481"},"maxAllocatedDisk":{},"consumedRu":{"valuePlusOne":"1"},"cpuTime":{"valuePlusOne":"0.000022405s"}},"output":{"numBatches":{"valuePlusOne":"2"},"numTuples":{"valuePlusOne":"2"}},"flowStats":{"maxMemUsage":{},"maxDiskUsage":{},"consumedRu":{}}}
              11.796ms      0.067ms            structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"00000000-0000-0000-0000-000000000000"},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{},"kvPairsRead":{},"tuplesRead":{},"batchRequestsIssued":{},"kvTime":{},"contentionTime":{},"lockWaitTime":{},"latchWaitTime":{},"numInterfaceSteps":{},"numInternalSteps":{},"numInterfaceSeeks":{},"numInternalSeeks":{},"blockBytes":{},"blockBytesInCache":{},"keyBytes":{},"valueBytes":{},"pointCount":{},"pointsCoveredByRangeTombstones":{},"rangeKeyCount":{},"rangeKeyContainedPoints":{},"rangeKeySkippedPoints":{},"kvCpuTime":{},"numGets":{},"numScans":{},"numReverseScans":{}},"exec":{"execTime":{},"maxAllocatedMem":{},"maxAllocatedDisk":{},"consumedRu":{},"cpuTime":{}},"output":{"numBatches":{},"numTuples":{}},"flowStats":{"maxMemUsage":{},"maxDiskUsage":{},"consumedRu":{}}}
              37.659ms     34.659ms        event:ComponentStats{ID: {3449d711-580b-425b-9508-e7d4251346ff FLOW 0 3 us-east-1}}
              37.182ms     -0.477ms        structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"3449d711-580b-425b-9508-e7d4251346ff","type":"FLOW","sqlInstanceId":3,"region":"us-east-1"},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{},"kvPairsRead":{},"tuplesRead":{},"batchRequestsIssued":{},"kvTime":{},"contentionTime":{},"lockWaitTime":{},"latchWaitTime":{},"numInterfaceSteps":{},"numInternalSteps":{},"numInterfaceSeeks":{},"numInternalSeeks":{},"blockBytes":{},"blockBytesInCache":{},"keyBytes":{},"valueBytes":{},"pointCount":{},"pointsCoveredByRangeTombstones":{},"rangeKeyCount":{},"rangeKeyContainedPoints":{},"rangeKeySkippedPoints":{},"kvCpuTime":{},"numGets":{},"numScans":{},"numReverseScans":{}},"exec":{"execTime":{},"maxAllocatedMem":{},"maxAllocatedDisk":{},"consumedRu":{},"cpuTime":{}},"output":{"numBatches":{},"numTuples":{}},"flowStats":{"maxMemUsage":{"valuePlusOne":"20481"},"maxDiskUsage":{"valuePlusOne":"1"},"consumedRu":{}}}
              37.796ms     34.986ms    event:sql/conn_executor_exec.go:1071 [n3,client=127.0.0.1:41026,hostssl,user=root] execution ends
              37.827ms      0.032ms    event:sql/conn_executor_exec.go:1071 [n3,client=127.0.0.1:41026,hostssl,user=root] rows affected: 1
              37.907ms      0.080ms        === operation:commit sql txn gid:83002 _verbose:1 node:3 client:127.0.0.1:41026 hostssl: user:root
              37.969ms      0.141ms    event:sql/conn_executor_exec.go:4071 [n3,client=127.0.0.1:41026,hostssl,user=root] AutoCommit. err: <nil>
        
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1a0d7b7d86f8cdff97782c5fadc5ffe4/logTestMultiRegionDataDriven_global_tables130132956
--- FAIL: TestMultiRegionDataDriven_global_tables (65.05s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven_global_tables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/multiregionccl.TestMultiRegionDataDriven_global_tables [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20703328?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20703328?buildTab=artifacts#/) on master @ [3442039e62fa53cc374f4d35ff9798313a942c5c](https://github.com/cockroachdb/cockroach/commits/3442039e62fa53cc374f4d35ff9798313a942c5c):


```
               5.533ms      0.007ms                            === operation:/cockroach.roachpb.Internal/Batch gid:111898 _verbose:1 node:4 span.kind:server request:Get [/Table/108/1/1/0], [txn: 88b844b9], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]
               5.550ms      0.017ms                            event:server/node.go:1622 [n4] node received request: 1 Get
               5.589ms      0.039ms                            event:kv/kvserver/store_send.go:149 [n4,s4] executing Get [/Table/108/1/1/0], [txn: 88b844b9], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]
               5.611ms      0.022ms                            event:kv/kvserver/replica_send.go:183 [n4,s4,r100/6:/{Table/108-Max}] read-only path
               5.633ms      0.022ms                            event:kv/kvserver/concurrency/concurrency_manager.go:275 [n4,s4,r100/6:/{Table/108-Max}] sequencing request
               5.649ms      0.016ms                            event:kv/kvserver/concurrency/concurrency_manager.go:356 [n4,s4,r100/6:/{Table/108-Max}] acquiring latches
               9.923ms      4.274ms                            event:kv/kvserver/concurrency/concurrency_manager.go:400 [n4,s4,r100/6:/{Table/108-Max}] scanning lock table for conflicting locks
               9.973ms      0.050ms                            event:kv/kvserver/replica_follower_read.go:133 [n4,s4,r100/6:/{Table/108-Max}] can't serve follower read; closed timestamp too low by: 77.484688ms; maxClosed: 1762438783.958897413,0 ts: 1762438783.536382101,0 uncertaintyLimit: 1762438784.036382101,0
              10.099ms      0.126ms                            event:kv/kvserver/replica_send.go:202 [n4,s4,r100/6:/{Table/108-Max}] replica.Send got error: [NotLeaseHolderError] lease held by different store; r100: replica (n4,s4):6NON_VOTER not lease holder; current lease is repl=(n1,s1):8VOTER_INCOMING seq=4 start=1762438776.228821664,0 type=LeaseEpoch epo=1 min-exp=1762438782.228776434,0 pro=1762438776.246833871,0 acq=Request
              10.190ms      0.090ms                            event:server/node.go:1690 [n4] error from stores.Send: [NotLeaseHolderError] lease held by different store; r100: replica (n4,s4):6NON_VOTER not lease holder; current lease is repl=(n1,s1):8VOTER_INCOMING seq=4 start=1762438776.228821664,0 type=LeaseEpoch epo=1 min-exp=1762438782.228776434,0 pro=1762438776.246833871,0 acq=Request: "sql txn" meta={id=88b844b9 key=/Min iso=Serializable pri=0.01653740 epo=0 ts=1762438783.536382101,0 min=1762438783.536382101,0 seq=0} lock=false stat=PENDING rts=1762438783.536382101,0 gul=1762438784.036382101,0 obs={n4@1762438783.536382101,0}
              10.210ms      0.020ms                            event:server/node.go:1727 [n4] node sending response
              10.285ms      4.759ms                        event:kv/kvclient/kvcoord/transport.go:212 [n4,client=127.0.0.1:52446,hostssl,user=root,txn=88b844b9] received batch response
              10.387ms      0.103ms                        event:kv/kvclient/kvcoord/dist_sender.go:2911 [n4,client=127.0.0.1:52446,hostssl,user=root,txn=88b844b9] [NotLeaseHolderError] lease held by different store; r100: replica (n4,s4):6NON_VOTER not lease holder; current lease is repl=(n1,s1):8VOTER_INCOMING seq=4 start=1762438776.228821664,0 type=LeaseEpoch epo=1 min-exp=1762438782.228776434,0 pro=1762438776.246833871,0 acq=Request: "sql txn" meta={id=88b844b9 key=/Min iso=Serializable pri=0.01653740 epo=0 ts=1762438783.536382101,0 min=1762438783.536382101,0 seq=0} lock=false stat=PENDING rts=1762438783.536382101,0 gul=1762438784.036382101,0 obs={n4@1762438783.536382101,0}
              10.481ms      0.093ms                        event:kv/kvclient/kvcoord/dist_sender.go:3161 [n4,client=127.0.0.1:52446,hostssl,user=root,txn=88b844b9] application error: [NotLeaseHolderError] lease held by different store; r100: replica (n4,s4):6NON_VOTER not lease holder; current lease is repl=(n1,s1):8VOTER_INCOMING seq=4 start=1762438776.228821664,0 type=LeaseEpoch epo=1 min-exp=1762438782.228776434,0 pro=1762438776.246833871,0 acq=Request: "sql txn" meta={id=88b844b9 key=/Min iso=Serializable pri=0.01653740 epo=0 ts=1762438783.536382101,0 min=1762438783.536382101,0 seq=0} lock=false stat=PENDING rts=1762438783.536382101,0 gul=1762438784.036382101,0 obs={n4@1762438783.536382101,0}
              10.514ms      0.033ms                        event:kv/kvclient/kvcoord/dist_sender.go:2695 [n4,client=127.0.0.1:52446,hostssl,user=root,txn=88b844b9] trying next peer (n1,s1):8
              10.541ms      0.027ms                        event:rpc/nodedialer/nodedialer.go:230 [n4,client=127.0.0.1:52446,hostssl,user=root,txn=88b844b9] sending request to 127.0.0.1:40471
              10.564ms      0.023ms                        event:kv/kvclient/kvcoord/transport.go:210 [n4,client=127.0.0.1:52446,hostssl,user=root,txn=88b844b9] sending batch request
              10.567ms      0.003ms                            === operation:/cockroach.roachpb.Internal/Batch gid:111898 _verbose:1 node:4 client:127.0.0.1:52446 hostssl: user:root txn:88b844b9 span.kind:client
              19.837ms      9.269ms                            === operation:/cockroach.roachpb.Internal/Batch gid:112790 _verbose:1 node:1 span.kind:server request:Get [/Table/108/1/1/0], [txn: 88b844b9], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]
              19.873ms      0.037ms                            event:server/node.go:1622 [n1] node received request: 1 Get
              19.935ms      0.061ms                            event:kv/kvserver/store_send.go:149 [n1,s1] executing Get [/Table/108/1/1/0], [txn: 88b844b9], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]
              19.964ms      0.029ms                            event:kv/kvserver/replica_send.go:183 [n1,s1,r100/8:/{Table/108-Max}] read-only path
              19.992ms      0.029ms                            event:kv/kvserver/concurrency/concurrency_manager.go:275 [n1,s1,r100/8:/{Table/108-Max}] sequencing request
              20.013ms      0.021ms                            event:kv/kvserver/concurrency/concurrency_manager.go:356 [n1,s1,r100/8:/{Table/108-Max}] acquiring latches
              20.043ms      0.029ms                            event:kv/kvserver/concurrency/concurrency_manager.go:400 [n1,s1,r100/8:/{Table/108-Max}] scanning lock table for conflicting locks
              20.201ms      0.158ms                            event:kv/kvserver/replica_read.go:317 [n1,s1,r100/8:/{Table/108-Max}] can drop latches early for batch (Get [/Table/108/1/1/0], [txn: 88b844b9], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]); scanning lock table first to detect conflicts
              20.266ms      0.065ms                            event:storage/engine.go:2090 [n1,s1,r100/8:/{Table/108-Max}] lock table scan stats: seeked 1 times (1 internal); stepped 0 times (0 internal)
              20.297ms      0.031ms                            event:kv/kvserver/replica_read.go:120 [n1,s1,r100/8:/{Table/108-Max}] lock table scan complete without conflicts; dropping latches early
              20.332ms      0.036ms                            event:kv/kvserver/replica_read.go:430 [n1,s1,r100/8:/{Table/108-Max}] executing read-only batch
              20.563ms      0.230ms                            event:kv/kvserver/replica_evaluate.go:567 [n1,s1,r100/8:/{Table/108-Max}] evaluated Get command header:<key:"\364\211\211\210" > expect_exclusion_since:<> , txn="sql txn" meta={id=88b844b9 key=/Min iso=Serializable pri=0.01653740 epo=0 ts=1762438783.536382101,0 min=1762438783.536382101,0 seq=0} lock=false stat=PENDING rts=1762438783.536382101,0 gul=1762438784.036382101,0 obs={n1@1762438783.556303660,0 n4@1762438783.536382101,0} : resp=header:<num_keys:1 num_bytes:7 > value:<raw_bytes:"\375\311[\010\n#\002" timestamp:<wall_time:1762438764472734919 > > , err=<nil>
              28.048ms      7.485ms                            event:n1 scan stats: stepped 0 times (0 internal); seeked 1 times (1 internal); block-bytes: (total 176 B, cached 176 B, duration 0s); points: (count 1, key-bytes 14 B, value-bytes 7 B, tombstoned: 0) ranges: (count 0), (contained-points 0, skipped-points 0) evaluated requests: 1 gets, 0 scans, 0 reverse scans
              28.092ms      0.044ms                            event:kv/kvserver/replica_read.go:242 [n1,s1,r100/8:/{Table/108-Max}] read completed
              28.159ms      0.068ms                            event:server/node.go:1727 [n1] node sending response
              20.574ms     -7.585ms                            structured:{"@type":"type.googleapis.com/cockroach.roachpb.ScanStats","numInterfaceSeeks":"1","numInternalSeeks":"1","blockBytes":"176","blockBytesInCache":"176","keyBytes":"14","valueBytes":"7","pointCount":"1","numGets":"1","nodeId":1,"region":"us-east-1"}
              36.573ms     26.009ms                        event:kv/kvclient/kvcoord/transport.go:212 [n4,client=127.0.0.1:52446,hostssl,user=root,txn=88b844b9] received batch response
              36.699ms     31.461ms                    event:kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:604 [n4,client=127.0.0.1:52446,hostssl,user=root,txn=88b844b9] recording span to refresh: /Table/108/1/1/0
              36.834ms     31.673ms            event:ComponentStats{ID: {69ce6893-5133-4907-ab72-7f5838def4c7 PROCESSOR 0 4 }, KV nodes: n1, KV regions: us-east-1, KV time: 32ms, KV rows decoded: 1, KV bytes read: 11 B, KV gRPC calls: 1, KV pairs read: 1, MVCC step count (ext/int): 0/0, MVCC seek count (ext/int): 1/1, max memory allocated: 20 KiB, sql cpu time: 25µs, batches output: 1, rows output: 1}
              36.853ms      0.019ms            event:ComponentStats{ID: {00000000-0000-0000-0000-000000000000 UNSET 0 ? }}
              36.769ms     -0.084ms            structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"69ce6893-5133-4907-ab72-7f5838def4c7","type":"PROCESSOR","sqlInstanceId":4},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{"valuePlusOne":"12"},"kvPairsRead":{"valuePlusOne":"2"},"tuplesRead":{"valuePlusOne":"2"},"batchRequestsIssued":{"valuePlusOne":"2"},"kvTime":{"valuePlusOne":"0.031584245s"},"contentionTime":{"valuePlusOne":"0.000000001s"},"lockWaitTime":{"valuePlusOne":"0.000000001s"},"latchWaitTime":{"valuePlusOne":"0.000000001s"},"numInterfaceSteps":{"valuePlusOne":"1"},"numInternalSteps":{"valuePlusOne":"1"},"numInterfaceSeeks":{"valuePlusOne":"2"},"numInternalSeeks":{"valuePlusOne":"2"},"blockBytes":{"valuePlusOne":"177"},"blockBytesInCache":{"valuePlusOne":"177"},"keyBytes":{"valuePlusOne":"15"},"valueBytes":{"valuePlusOne":"8"},"pointCount":{"valuePlusOne":"2"},"pointsCoveredByRangeTombstones":{"valuePlusOne":"1"},"rangeKeyCount":{"valuePlusOne":"1"},"rangeKeyContainedPoints":{"valuePlusOne":"1"},"rangeKeySkippedPoints":{"valuePlusOne":"1"},"kvCpuTime":{},"numGets":{"valuePlusOne":"2"},"numScans":{"valuePlusOne":"1"},"numReverseScans":{"valuePlusOne":"1"},"nodeIds":[1],"regions":["us-east-1"]},"exec":{"execTime":{},"maxAllocatedMem":{"valuePlusOne":"20481"},"maxAllocatedDisk":{},"consumedRu":{"valuePlusOne":"1"},"cpuTime":{"valuePlusOne":"0.000025274s"}},"output":{"numBatches":{"valuePlusOne":"2"},"numTuples":{"valuePlusOne":"2"}},"flowStats":{"maxMemUsage":{},"maxDiskUsage":{},"consumedRu":{}}}
              36.839ms      0.069ms            structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"00000000-0000-0000-0000-000000000000"},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{},"kvPairsRead":{},"tuplesRead":{},"batchRequestsIssued":{},"kvTime":{},"contentionTime":{},"lockWaitTime":{},"latchWaitTime":{},"numInterfaceSteps":{},"numInternalSteps":{},"numInterfaceSeeks":{},"numInternalSeeks":{},"blockBytes":{},"blockBytesInCache":{},"keyBytes":{},"valueBytes":{},"pointCount":{},"pointsCoveredByRangeTombstones":{},"rangeKeyCount":{},"rangeKeyContainedPoints":{},"rangeKeySkippedPoints":{},"kvCpuTime":{},"numGets":{},"numScans":{},"numReverseScans":{}},"exec":{"execTime":{},"maxAllocatedMem":{},"maxAllocatedDisk":{},"consumedRu":{},"cpuTime":{}},"output":{"numBatches":{},"numTuples":{}},"flowStats":{"maxMemUsage":{},"maxDiskUsage":{},"consumedRu":{}}}
              36.922ms     31.760ms        event:ComponentStats{ID: {69ce6893-5133-4907-ab72-7f5838def4c7 FLOW 0 4 us-west-1}}
              36.906ms     -0.016ms        structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"69ce6893-5133-4907-ab72-7f5838def4c7","type":"FLOW","sqlInstanceId":4,"region":"us-west-1"},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{},"kvPairsRead":{},"tuplesRead":{},"batchRequestsIssued":{},"kvTime":{},"contentionTime":{},"lockWaitTime":{},"latchWaitTime":{},"numInterfaceSteps":{},"numInternalSteps":{},"numInterfaceSeeks":{},"numInternalSeeks":{},"blockBytes":{},"blockBytesInCache":{},"keyBytes":{},"valueBytes":{},"pointCount":{},"pointsCoveredByRangeTombstones":{},"rangeKeyCount":{},"rangeKeyContainedPoints":{},"rangeKeySkippedPoints":{},"kvCpuTime":{},"numGets":{},"numScans":{},"numReverseScans":{}},"exec":{"execTime":{},"maxAllocatedMem":{},"maxAllocatedDisk":{},"consumedRu":{},"cpuTime":{}},"output":{"numBatches":{},"numTuples":{}},"flowStats":{"maxMemUsage":{"valuePlusOne":"20481"},"maxDiskUsage":{"valuePlusOne":"1"},"consumedRu":{}}}
              37.027ms     31.981ms    event:sql/conn_executor_exec.go:2079 [n4,client=127.0.0.1:52446,hostssl,user=root] execution ends
              37.072ms      0.045ms    event:sql/conn_executor_exec.go:2079 [n4,client=127.0.0.1:52446,hostssl,user=root] rows affected: 1
              37.153ms      0.081ms        === operation:commit sql txn gid:111898 _verbose:1 node:4 client:127.0.0.1:52446 hostssl: user:root
              37.214ms      0.142ms    event:sql/conn_executor_exec.go:4122 [n4,client=127.0.0.1:52446,hostssl,user=root] AutoCommit. err: <nil>
        
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1a0d7b7d86f8cdff97782c5fadc5ffe4/logTestMultiRegionDataDriven_global_tables1077515884
--- FAIL: TestMultiRegionDataDriven_global_tables (87.02s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven_global_tables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/multiregionccl.TestMultiRegionDataDriven_global_tables [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20765951?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20765951?buildTab=artifacts#/) on master @ [ab5107e24df2f9ece006b5e7519541e1e38c7766](https://github.com/cockroachdb/cockroach/commits/ab5107e24df2f9ece006b5e7519541e1e38c7766):


```
               1.192ms      0.014ms                        event:kv/kvclient/kvcoord/transport.go:210 [n6,client=127.0.0.1:34764,hostssl,user=root,txn=27d16659] sending batch request
               1.201ms      0.010ms                            === operation:/cockroach.roachpb.Internal/Batch gid:93408 _verbose:1 node:6 span.kind:server request:Get [/Table/108/1/1/0], [txn: 27d16659], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]
               1.212ms      0.011ms                            event:server/node.go:1617 [n6] node received request: 1 Get
               1.244ms      0.031ms                            event:kv/kvserver/store_send.go:149 [n6,s6] executing Get [/Table/108/1/1/0], [txn: 27d16659], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]
               1.261ms      0.017ms                            event:kv/kvserver/replica_send.go:183 [n6,s6,r90/10:/{Table/108-Max}] read-only path
               1.279ms      0.018ms                            event:kv/kvserver/concurrency/concurrency_manager.go:275 [n6,s6,r90/10:/{Table/108-Max}] sequencing request
               1.290ms      0.011ms                            event:kv/kvserver/concurrency/concurrency_manager.go:356 [n6,s6,r90/10:/{Table/108-Max}] acquiring latches
               1.303ms      0.013ms                            event:kv/kvserver/concurrency/concurrency_manager.go:400 [n6,s6,r90/10:/{Table/108-Max}] scanning lock table for conflicting locks
               1.338ms      0.035ms                            event:kv/kvserver/replica_follower_read.go:133 [n6,s6,r90/10:/{Table/108-Max}] can't serve follower read; closed timestamp too low by: 9.046947ms; maxClosed: 1763746012.361655332,0 ts: 1763746011.870702279,0 uncertaintyLimit: 1763746012.370702279,0
               1.441ms      0.103ms                            event:kv/kvserver/replica_send.go:202 [n6,s6,r90/10:/{Table/108-Max}] replica.Send got error: [NotLeaseHolderError] lease held by different store; r90: replica (n6,s6):10NON_VOTER not lease holder; current lease is repl=(n1,s1):8 seq=3 start=1763745999.247119472,0 type=LeaseExpiration exp=1763746015.239666108,0 pro=1763746009.239666108,0 acq=Request
               1.513ms      0.072ms                            event:server/node.go:1685 [n6] error from stores.Send: [NotLeaseHolderError] lease held by different store; r90: replica (n6,s6):10NON_VOTER not lease holder; current lease is repl=(n1,s1):8 seq=3 start=1763745999.247119472,0 type=LeaseExpiration exp=1763746015.239666108,0 pro=1763746009.239666108,0 acq=Request: "sql txn" meta={id=27d16659 key=/Min iso=Serializable pri=0.01316899 epo=0 ts=1763746011.870702279,0 min=1763746011.870702279,0 seq=0} lock=false stat=PENDING rts=1763746011.870702279,0 gul=1763746012.370702279,0 obs={n6@1763746011.870702279,0}
               1.528ms      0.015ms                            event:server/node.go:1722 [n6] node sending response
               1.572ms      0.380ms                        event:kv/kvclient/kvcoord/transport.go:212 [n6,client=127.0.0.1:34764,hostssl,user=root,txn=27d16659] received batch response
               1.630ms      0.058ms                        event:kv/kvclient/kvcoord/dist_sender.go:2911 [n6,client=127.0.0.1:34764,hostssl,user=root,txn=27d16659] [NotLeaseHolderError] lease held by different store; r90: replica (n6,s6):10NON_VOTER not lease holder; current lease is repl=(n1,s1):8 seq=3 start=1763745999.247119472,0 type=LeaseExpiration exp=1763746015.239666108,0 pro=1763746009.239666108,0 acq=Request: "sql txn" meta={id=27d16659 key=/Min iso=Serializable pri=0.01316899 epo=0 ts=1763746011.870702279,0 min=1763746011.870702279,0 seq=0} lock=false stat=PENDING rts=1763746011.870702279,0 gul=1763746012.370702279,0 obs={n6@1763746011.870702279,0}
               1.696ms      0.066ms                        event:kv/kvclient/kvcoord/dist_sender.go:3161 [n6,client=127.0.0.1:34764,hostssl,user=root,txn=27d16659] application error: [NotLeaseHolderError] lease held by different store; r90: replica (n6,s6):10NON_VOTER not lease holder; current lease is repl=(n1,s1):8 seq=3 start=1763745999.247119472,0 type=LeaseExpiration exp=1763746015.239666108,0 pro=1763746009.239666108,0 acq=Request: "sql txn" meta={id=27d16659 key=/Min iso=Serializable pri=0.01316899 epo=0 ts=1763746011.870702279,0 min=1763746011.870702279,0 seq=0} lock=false stat=PENDING rts=1763746011.870702279,0 gul=1763746012.370702279,0 obs={n6@1763746011.870702279,0}
               1.719ms      0.023ms                        event:kv/kvclient/kvcoord/dist_sender.go:2695 [n6,client=127.0.0.1:34764,hostssl,user=root,txn=27d16659] trying next peer (n1,s1):8
               1.737ms      0.019ms                        event:rpc/nodedialer/nodedialer.go:230 [n6,client=127.0.0.1:34764,hostssl,user=root,txn=27d16659] sending request to 127.0.0.1:39791
               1.754ms      0.017ms                        event:kv/kvclient/kvcoord/transport.go:210 [n6,client=127.0.0.1:34764,hostssl,user=root,txn=27d16659] sending batch request
               1.902ms      0.147ms                            === operation:/cockroach.roachpb.Internal/Batch gid:83070 _verbose:1 node:1 span.kind:server request:Get [/Table/108/1/1/0], [txn: 27d16659], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]
               1.922ms      0.020ms                            event:server/node.go:1617 [n1] node received request: 1 Get
               1.960ms      0.038ms                            event:kv/kvserver/store_send.go:149 [n1,s1] executing Get [/Table/108/1/1/0], [txn: 27d16659], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]
               1.976ms      0.016ms                            event:kv/kvserver/replica_send.go:183 [n1,s1,r90/8:/{Table/108-Max}] read-only path
               1.996ms      0.020ms                            event:kv/kvserver/concurrency/concurrency_manager.go:275 [n1,s1,r90/8:/{Table/108-Max}] sequencing request
               2.006ms      0.010ms                            event:kv/kvserver/concurrency/concurrency_manager.go:356 [n1,s1,r90/8:/{Table/108-Max}] acquiring latches
               2.019ms      0.013ms                            event:kv/kvserver/concurrency/concurrency_manager.go:400 [n1,s1,r90/8:/{Table/108-Max}] scanning lock table for conflicting locks
               2.123ms      0.104ms                            event:kv/kvserver/replica_read.go:317 [n1,s1,r90/8:/{Table/108-Max}] can drop latches early for batch (Get [/Table/108/1/1/0], [txn: 27d16659], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 10485760]); scanning lock table first to detect conflicts
               2.169ms      0.046ms                            event:storage/engine.go:2017 [n1,s1,r90/8:/{Table/108-Max}] lock table scan stats: seeked 1 times (1 internal); stepped 0 times (0 internal)
               2.181ms      0.012ms                            event:kv/kvserver/replica_read.go:120 [n1,s1,r90/8:/{Table/108-Max}] lock table scan complete without conflicts; dropping latches early
               2.204ms      0.022ms                            event:kv/kvserver/replica_read.go:430 [n1,s1,r90/8:/{Table/108-Max}] executing read-only batch
               2.366ms      0.163ms                            event:kv/kvserver/replica_evaluate.go:567 [n1,s1,r90/8:/{Table/108-Max}] evaluated Get command header:<key:"\364\211\211\210" > expect_exclusion_since:<> , txn="sql txn" meta={id=27d16659 key=/Min iso=Serializable pri=0.01316899 epo=0 ts=1763746011.870702279,0 min=1763746011.870702279,0 seq=0} lock=false stat=PENDING rts=1763746011.870702279,0 gul=1763746012.370702279,0 obs={n1@1763746011.872710651,0 n6@1763746011.870702279,0} : resp=header:<num_keys:1 num_bytes:7 > value:<raw_bytes:"\375\311[\010\n#\002" timestamp:<wall_time:1763745983678183243 > > , err=<nil>
               2.393ms      0.027ms                            event:n1 scan stats: stepped 0 times (0 internal); seeked 1 times (1 internal); block-bytes: (total 190 B, cached 190 B, duration 0s); points: (count 1, key-bytes 14 B, value-bytes 14 B, tombstoned: 0) ranges: (count 0), (contained-points 0, skipped-points 0) evaluated requests: 1 gets, 0 scans, 0 reverse scans
               2.409ms      0.015ms                            event:kv/kvserver/replica_read.go:242 [n1,s1,r90/8:/{Table/108-Max}] read completed
               2.450ms      0.042ms                            event:server/node.go:1722 [n1] node sending response
               2.374ms     -0.077ms                            structured:{"@type":"type.googleapis.com/cockroach.roachpb.ScanStats","numInterfaceSeeks":"1","numInternalSeeks":"1","blockBytes":"190","blockBytesInCache":"190","keyBytes":"14","valueBytes":"14","pointCount":"1","numGets":"1","nodeId":1,"region":"us-east-1"}
              28.690ms     26.936ms                        event:kv/kvclient/kvcoord/transport.go:212 [n6,client=127.0.0.1:34764,hostssl,user=root,txn=27d16659] received batch response
              28.771ms     27.812ms                    event:kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:617 [n6,client=127.0.0.1:34764,hostssl,user=root,txn=27d16659] recording span to refresh: /Table/108/1/1/0
              28.871ms     27.978ms            event:ComponentStats{ID: {862d4bce-51b2-46ef-bd02-906734e3eea1 PROCESSOR 0 6 }, KV nodes: n1, KV regions: us-east-1, KV time: 28ms, KV rows decoded: 1, KV bytes read: 11 B, KV gRPC calls: 1, KV pairs read: 1, MVCC step count (ext/int): 0/0, MVCC seek count (ext/int): 1/1, max memory allocated: 20 KiB, sql cpu time: 25µs, batches output: 1, rows output: 1}
              28.879ms      0.009ms            event:ComponentStats{ID: {00000000-0000-0000-0000-000000000000 UNSET 0 ? }}
              28.834ms     -0.045ms            structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"862d4bce-51b2-46ef-bd02-906734e3eea1","type":"PROCESSOR","sqlInstanceId":6},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{"valuePlusOne":"12"},"kvPairsRead":{"valuePlusOne":"2"},"tuplesRead":{"valuePlusOne":"2"},"batchRequestsIssued":{"valuePlusOne":"2"},"kvTime":{"valuePlusOne":"0.027915863s"},"contentionTime":{"valuePlusOne":"0.000000001s"},"lockWaitTime":{"valuePlusOne":"0.000000001s"},"latchWaitTime":{"valuePlusOne":"0.000000001s"},"numInterfaceSteps":{"valuePlusOne":"1"},"numInternalSteps":{"valuePlusOne":"1"},"numInterfaceSeeks":{"valuePlusOne":"2"},"numInternalSeeks":{"valuePlusOne":"2"},"blockBytes":{"valuePlusOne":"191"},"blockBytesInCache":{"valuePlusOne":"191"},"keyBytes":{"valuePlusOne":"15"},"valueBytes":{"valuePlusOne":"15"},"pointCount":{"valuePlusOne":"2"},"pointsCoveredByRangeTombstones":{"valuePlusOne":"1"},"rangeKeyCount":{"valuePlusOne":"1"},"rangeKeyContainedPoints":{"valuePlusOne":"1"},"rangeKeySkippedPoints":{"valuePlusOne":"1"},"kvCpuTime":{},"numGets":{"valuePlusOne":"2"},"numScans":{"valuePlusOne":"1"},"numReverseScans":{"valuePlusOne":"1"},"nodeIds":[1],"regions":["us-east-1"]},"exec":{"execTime":{},"maxAllocatedMem":{"valuePlusOne":"20481"},"maxAllocatedDisk":{},"consumedRu":{"valuePlusOne":"1"},"cpuTime":{"valuePlusOne":"0.000025273s"}},"output":{"numBatches":{"valuePlusOne":"2"},"numTuples":{"valuePlusOne":"2"}},"flowStats":{"maxMemUsage":{},"maxDiskUsage":{},"consumedRu":{}}}
              28.873ms      0.039ms            structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"00000000-0000-0000-0000-000000000000"},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{},"kvPairsRead":{},"tuplesRead":{},"batchRequestsIssued":{},"kvTime":{},"contentionTime":{},"lockWaitTime":{},"latchWaitTime":{},"numInterfaceSteps":{},"numInternalSteps":{},"numInterfaceSeeks":{},"numInternalSeeks":{},"blockBytes":{},"blockBytesInCache":{},"keyBytes":{},"valueBytes":{},"pointCount":{},"pointsCoveredByRangeTombstones":{},"rangeKeyCount":{},"rangeKeyContainedPoints":{},"rangeKeySkippedPoints":{},"kvCpuTime":{},"numGets":{},"numScans":{},"numReverseScans":{}},"exec":{"execTime":{},"maxAllocatedMem":{},"maxAllocatedDisk":{},"consumedRu":{},"cpuTime":{}},"output":{"numBatches":{},"numTuples":{}},"flowStats":{"maxMemUsage":{},"maxDiskUsage":{},"consumedRu":{}}}
              28.927ms     28.035ms        event:ComponentStats{ID: {862d4bce-51b2-46ef-bd02-906734e3eea1 FLOW 0 6 eu-central-1}}
              28.918ms     -0.008ms        structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"862d4bce-51b2-46ef-bd02-906734e3eea1","type":"FLOW","sqlInstanceId":6,"region":"eu-central-1"},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{},"kvPairsRead":{},"tuplesRead":{},"batchRequestsIssued":{},"kvTime":{},"contentionTime":{},"lockWaitTime":{},"latchWaitTime":{},"numInterfaceSteps":{},"numInternalSteps":{},"numInterfaceSeeks":{},"numInternalSeeks":{},"blockBytes":{},"blockBytesInCache":{},"keyBytes":{},"valueBytes":{},"pointCount":{},"pointsCoveredByRangeTombstones":{},"rangeKeyCount":{},"rangeKeyContainedPoints":{},"rangeKeySkippedPoints":{},"kvCpuTime":{},"numGets":{},"numScans":{},"numReverseScans":{}},"exec":{"execTime":{},"maxAllocatedMem":{},"maxAllocatedDisk":{},"consumedRu":{},"cpuTime":{}},"output":{"numBatches":{},"numTuples":{}},"flowStats":{"maxMemUsage":{"valuePlusOne":"20481"},"maxDiskUsage":{"valuePlusOne":"1"},"consumedRu":{}}}
              28.990ms     28.213ms    event:sql/conn_executor_exec.go:1081 [n6,client=127.0.0.1:34764,hostssl,user=root] execution ends
              29.011ms      0.021ms    event:sql/conn_executor_exec.go:1081 [n6,client=127.0.0.1:34764,hostssl,user=root] rows affected: 1
              29.074ms      0.064ms        === operation:commit sql txn gid:93408 _verbose:1 node:6 client:127.0.0.1:34764 hostssl: user:root
              29.112ms      0.101ms    event:sql/conn_executor_exec.go:4119 [n6,client=127.0.0.1:34764,hostssl,user=root] AutoCommit. err: <nil>
        
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/92210d9178e507c2b00f19c1ad4bbd36/logTestMultiRegionDataDriven_global_tables3118765647
--- FAIL: TestMultiRegionDataDriven_global_tables (81.74s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionDataDriven_global_tables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/multiregionccl/datadriven_test.go
+++ b/pkg/ccl/multiregionccl/datadriven_test.go
@@ -261,10 +261,13 @@ SET CLUSTER SETTING kv.allocator.min_lease_transfer_interval = '5m'
 
 			case "trace-sql":
 				mustHaveArgOrFatal(t, d, serverIdx)
+				var idx int
+				d.ScanArgs(t, serverIdx, &idx)
+				shouldRetry := d.HasArg(traceSQLRetryArg)
+				retryTimeout := testutils.DefaultSucceedsSoonDuration
+
 				var rec tracingpb.Recording
 				queryFunc := func() (localRead bool, followerRead bool, err error) {
-					var idx int
-					d.ScanArgs(t, serverIdx, &idx)
 					sqlDB, err := ds.getSQLConn(idx)
 					if err != nil {
 						return false, false, err
@@ -291,22 +294,41 @@ SET CLUSTER SETTING kv.allocator.min_lease_transfer_interval = '5m'
 					}
 					return localRead, followerRead, nil
 				}
-				localRead, followerRead, err := queryFunc()
+
+				runOnce := func() (string, error) {
+					localRead, followerRead, err := queryFunc()
+					if err != nil {
+						return "", err
+					}
+					var output strings.Builder
+					output.WriteString(
+						fmt.Sprintf("served locally: %s\n", strconv.FormatBool(localRead)))
+					// Only print follower read information if the query was served locally.
+					if localRead {
+						output.WriteString(
+							fmt.Sprintf("served via follower read: %s\n", strconv.FormatBool(followerRead)))
+					}
+					if d.Expected != output.String() {
+						return "", errors.AssertionFailedf("not a match, trace:\n%s\n", rec)
+					}
+					return output.String(), nil
+				}
+
+				var output string
+				var err error
+				if shouldRetry {
+					err = testutils.SucceedsWithinError(func() error {
+						var attemptErr error
+						output, attemptErr = runOnce()
+						return attemptErr
+					}, retryTimeout)
+				} else {
+					output, err = runOnce()
+				}
 				if err != nil {
 					return err.Error()
 				}
-				var output strings.Builder
-				output.WriteString(
-					fmt.Sprintf("served locally: %s\n", strconv.FormatBool(localRead)))
-				// Only print follower read information if the query was served locally.
-				if localRead {
-					output.WriteString(
-						fmt.Sprintf("served via follower read: %s\n", strconv.FormatBool(followerRead)))
-				}
-				if d.Expected != output.String() {
-					return errors.AssertionFailedf("not a match, trace:\n%s\n", rec).Error()
-				}
-				return output.String()
+				return output
 
 			case "refresh-range-descriptor-cache":
 				mustHaveArgOrFatal(t, d, tableName)
@@ -484,6 +506,7 @@ const (
 	partitionName    = "partition-name"
 	numVoters        = "num-voters"
 	numNonVoters     = "num-non-voters"
+	traceSQLRetryArg = "retry"
 )
 
 type replicaType int

--- a/pkg/ccl/multiregionccl/testdata/global_tables
+++ b/pkg/ccl/multiregionccl/testdata/global_tables
@@ -32,7 +32,7 @@ SELECT * FROM db.global WHERE k = 2
 ----
 LEAD_FOR_GLOBAL_READS
 
-trace-sql idx=1
+trace-sql idx=1 retry
 SELECT * FROM db.global WHERE k = 1
 ----
 served locally: true
@@ -43,7 +43,7 @@ SELECT * FROM db.global WHERE k = 2
 ----
 LEAD_FOR_GLOBAL_READS
 
-trace-sql idx=2
+trace-sql idx=2 retry
 SELECT * FROM db.global WHERE k = 1
 ----
 served locally: true
@@ -54,7 +54,7 @@ SELECT * FROM db.global WHERE k = 2
 ----
 LEAD_FOR_GLOBAL_READS
 
-trace-sql idx=3
+trace-sql idx=3 retry
 SELECT * FROM db.global WHERE k = 1
 ----
 served locally: true
@@ -65,7 +65,7 @@ SELECT * FROM db.global WHERE k = 2
 ----
 LEAD_FOR_GLOBAL_READS
 
-trace-sql idx=4
+trace-sql idx=4 retry
 SELECT * FROM db.global WHERE k = 1
 ----
 served locally: true
@@ -95,7 +95,7 @@ SELECT * FROM db.global WHERE k = 2
 ----
 LEAD_FOR_GLOBAL_READS
 
-trace-sql idx=5
+trace-sql idx=5 retry
 SELECT * FROM db.global WHERE k = 1
 ----
 served locally: true

